libvalkeycluster/valkeycluster.h

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright (c) 2015-2017, Ieshen Zheng <ieshen.zheng at 163 dot com>
+ * Copyright (c) 2020, Nick <heronr1 at gmail dot com>
+ * Copyright (c) 2020-2021, Bjorn Svensson <bjorn.a.svensson at est dot tech>
+ * Copyright (c) 2021, Red Hat
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *   * Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of Redis nor the names of its contributors may be used
+ *     to endorse or promote products derived from this software without
+ *     specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef __HIRCLUSTER_H
+#define __HIRCLUSTER_H
+
+#include "dict.h"
+#include <valkey/async.h>
+#include <valkey/valkey.h>
+
+#define UNUSED(x) (void)(x)
+
+#define LIBVALKEYCLUSTER_MAJOR 0
+#define LIBVALKEYCLUSTER_MINOR 14
+#define LIBVALKEYCLUSTER_PATCH 0
+#define LIBVALKEYCLUSTER_SONAME 0.14
+
+#define VALKEYCLUSTER_SLOTS 16384
+
+#define VALKEY_ROLE_NULL 0
+#define VALKEY_ROLE_MASTER 1
```

Comment:
Can we change the terminology?

Reply:
Sure, the master/slave terminology should be removed

Reply:
Create follow-up issue?

Reply:
Yes, separate PR